build-ps/jenkins/build-binary

```diff
@@ -0,0 +1,164 @@
+fi
+if [[ $(getconf LONG_BIT) == 32 ]]; then
+    TARGET_ARCH="i686"
+    TARGET_CFLAGS+=" -m32 -march=i686"
+else
+    TARGET_ARCH="$(uname -m)"
+fi
+
+
+# ------------------------------------------------------------------------------
+# Set Debug options
+# ------------------------------------------------------------------------------
+if [[ $CMAKE_BUILD_TYPE == Debug ]]; then
+    BUILD_COMMENT+="-debug"
+    TARGET_CFLAGS=$(
+        echo " ${TARGET_CFLAGS} " | \
+            sed -e 's/ -O[0-9]* / /' \
+                -e 's/-Wp,-D_FORTIFY_SOURCE=2/ /' \
+                -e 's/ -unroll2 / /' \
+                -e 's/ -ip / /'
+    )
+    CMAKE_OPT+=" -DMYSQL_MAINTAINER_MODE=ON -DDEBUG_EXTNAME=OFF"
+else
+    CMAKE_OPT+=" -DMYSQL_MAINTAINER_MODE=OFF"
+fi
+if [[ ${CMAKE_OPTS} == *WITH_VALGRIND=ON* ]]; then
+    BUILD_COMMENT+="-valgrind"
+fi
+
+
+# ------------------------------------------------------------------------------
+# set version
+# ------------------------------------------------------------------------------
+source "$SOURCEDIR/VERSION"
+if [[ -n $(which git) ]] && [[ -d "$SOURCEDIR/.git" ]]; then
+    REVISION="$(cd "$SOURCEDIR"; git rev-parse --short HEAD)"
+fi
+MYSQL_VERSION="$MYSQL_VERSION_MAJOR.$MYSQL_VERSION_MINOR.$MYSQL_VERSION_PATCH"
+PERCONA_SERVER_VERSION="$(echo $MYSQL_VERSION_EXTRA | sed 's/^-//')"
+PRODUCT="Percona-Server-$MYSQL_VERSION-$PERCONA_SERVER_VERSION"
+TOKUDB_BACKUP_VERSION="${MYSQL_VERSION}${MYSQL_VERSION_EXTRA}"
+PRODUCT_FULL="Percona-Server-${MYSQL_VERSION}-${PERCONA_SERVER_VERSION}${BUILD_COMMENT}-${TAG}$(uname -s)${DIST_NAME}.${TARGET_ARCH}${SSL_VER}"
+COMMENT="Percona Server (GPL), Release ${MYSQL_VERSION_EXTRA#-}, Revision ${REVISION:-}${BUILD_COMMENT}"
+
+
+# ------------------------------------------------------------------------------
+# Check tokudb
+# ------------------------------------------------------------------------------
+if [[ ${WITH_TOKUDB} == ON ]]; then
+    CMAKE_OPTS+=" -DBUILD_TESTING=OFF -DUSE_GTAGS=OFF -DUSE_CTAGS=OFF -DUSE_ETAGS=OFF -DUSE_CSCOPE=OFF -DTOKUDB_BACKUP_PLUGIN_VERSION=${TOKUDB_BACKUP_VERSION}"
+    if [[ ${CMAKE_BUILD_TYPE} == Debug ]]; then
+        CMAKE_OPTS+=" -DTOKU_DEBUG_PARANOID=ON"
+    else
+        CMAKE_OPTS+=" -DTOKU_DEBUG_PARANOID=OFF"
+    fi
+    if [[ $CMAKE_OPTS == *WITH_VALGRIND=ON* ]]; then
+        CMAKE_OPTS+=" -DUSE_VALGRIND=ON"
+    fi
+fi
+
+
+# ------------------------------------------------------------------------------
+# Check mecab
+# ------------------------------------------------------------------------------
+if [ -f /etc/mecabrc ]; then
+    WITH_MECAB="system"
+else
+    WITH_MECAB=""
+fi
+
+
+# ------------------------------------------------------------------------------
+# Set compile options
+# ------------------------------------------------------------------------------
+if [[ $COMPILER != default ]]; then
+    export CC=${COMPILER}
+    export CXX=${COMPILER/cc/++}
```

Comment:
this does not work for `clang`:
`gcc` -> `g++`
`clang` -> `clang++`

Reply:
improved